### PR TITLE
raftstore: support changing label key for replication

### PR DESCRIPTION
**Pull request by BusyJay**
### What is changed and how it works?

This PR supports changing label key at runtime.

PD may not specify label key until replication mode is DR_AUTO_SYNC. Even replication mode is configured, label key can be wrong and need to be corrected. So loading label value during starting is not a feasible solution.

This PR caches all label key-value instead. So even PD change label key, new group ID can be still be re-calculated instead of loading all from PD again.

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

- No release node.

### CODE REVIEW

**Comment by NingLin-P** [comment]
Should we also support the change of each store's labels?

File: components/raftstore/src/store/fsm/peer.rs [hunk 1]
```diff
@@ -2402,8 +2403,10 @@ impl<'a, T: Transport, C: PdClient> PeerFsmDelegate<'a, T, C> {
 
         if prev_region.get_peers() != region.get_peers() {
             let mut state = self.ctx.global_replication_state.lock().unwrap();
-            let gb = state.calculate_commit_group(region.get_peers());
-            self.fsm.peer.raft_group.raft.assign_commit_groups(&gb);
+            let gb = state
+                .calculate_commit_group(self.fsm.peer.replication_mode_version, region.get_peers());
+            self.fsm.peer.raft_group.raft.clear_commit_group();
+            self.fsm.peer.raft_group.raft.assign_commit_groups(gb);
```
[the remark this answers is not in the document]
Comment (BusyJay) [reply components/raftstore/src/store/fsm/peer.rs:2409]: Empty result is OK, it should not affect correctness.

File: components/raftstore/src/store/fsm/peer.rs [hunk 2]
```diff
@@ -1703,13 +1703,14 @@ impl<'a, T: Transport, C: PdClient> PeerFsmDelegate<'a, T, C> {
                     .lock()
                     .unwrap()
                     .group
-                    .group_id(peer.store_id);
-                if let Some(group_id) = group_id {
+                    .group_id(self.fsm.peer.replication_mode_version, peer.store_id);
```
[the remark this answers is not in the document]
Comment (BusyJay) [reply components/raftstore/src/store/fsm/peer.rs:1706]: When `replication_mode_version` is updated, the commit group is recalculated.

**Comment by BusyJay** [comment]
No, we don't support label value update.

**Pull request merged**